src/Codeception/Module/WebDriver.php

```diff
@@ -2867,12 +2867,21 @@ protected function convertKeyModifier($keys)
 
     protected function assertNodesContain($text, $nodes, $selector = null)
     {
-        $this->assertThat($nodes, new WebDriverConstraint($text, $this->_getCurrentUri()), $selector);
+        $this->assertNodeConstraint($nodes, new WebDriverConstraint($text, $this->_getCurrentUri()), $selector);
     }
 
     protected function assertNodesNotContain($text, $nodes, $selector = null)
     {
-        $this->assertThat($nodes, new WebDriverConstraintNot($text, $this->_getCurrentUri()), $selector);
+        $this->assertNodeConstraint($nodes, new WebDriverConstraintNot($text, $this->_getCurrentUri()), $selector);
+    }
+
+    protected function assertNodeConstraint($nodes, WebDriverConstraint $constraint, $selector = null)
+    {
+        $message = $selector;
+        if (is_array($selector)) {
+            $message = implode(array_keys($selector)) . ':' . implode($selector);
```

Comment:
Result of implode without separator can be confusing if $selector array contains multiple items.
I think that it would be better to replicate behaviour of [getStrictLocator](https://github.com/Codeception/Codeception/blob/2.4.1/src/Codeception/Module/WebDriver.php#L2771-L2772) method
```php
$type = key($selector);
$locator = $selector[$type];
```

Reply:
@edno have you got any objections to proposed change?